wasm-node/javascript/src/internals/client.ts

```diff
@@ -552,27 +552,45 @@ export function start(options: ClientOptions, wasmModule: SmoldotBytecode | Prom
                         default: throw new Error("Internal error: unknown json_rpc_send error code: " + retVal)
                     }
                 },
+                jsonRpcResponses: {
+                    next: async () => {
+                        while (true) {
+                            if (!state.chains.has(chainId))
+                                return { done: true, value: undefined };
+                            if (options.disableJsonRpc)
+                                throw new JsonRpcDisabledError();
+                            if (state.instance.status === "destroyed")
+                                throw state.instance.error;
+                            if (state.instance.status !== "ready")
+                                throw new Error(); // Internal error. Never supposed to happen.
+    
+                            // Try to pop a message from the queue.
+                            const message = state.instance.instance.peekJsonRpcResponse(chainId);
+                            if (message)
+                                return { done: false, value: message };
+    
+                            // If no message is available, wait for one to be.
+                            await new Promise<void>((resolve) => {
+                                state.chains.get(chainId)!.jsonRpcResponsesPromises.push(resolve)
+                            });
+                        }
+                    },
+                    return: async () => {
+                        newChain.remove();
```

Comment:
Why remove the chain here?

Reply:
I believe this behaviour make sense according to the documentation [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#returnvalue). But I have removed it anyway, let me know what is your final judgement on this, thank you.